bundle/edu.gemini.catalog/src/main/java/jsky/catalog/AbstractCatalogDirectory.java

```diff
@@ -582,7 +549,7 @@ private TreeModelEvent _getTreeModelEvent(Catalog cat) {
         int n = l.size();
         Catalog[] ar = new Catalog[n];
         for (int i = 0; i < n; i++)
-            ar[n - i - 1] = (Catalog) l.get(i);
+            ar[n - i - 1] = l.get(i);
```

Comment:
As long as we're cleaning things up, we could make `l` an `ArrayList` and then something like this:

```java
Collections.reverse(l);
return l.toArray(new Catalog[l.size()]);
```